### PR TITLE
Emit with Callback Node, Connection Auth added and updated packages

### PR DESCRIPTION
**Pull request by nateainsworth**
Added:
- Node with callbacks on Emit.
- Auth added to config for JWT middleware securing connections.
- Port validation updated to use regex to accept a blank field or numbers.
- Updated socekt-io-client version.
- Placeholder notes to make it clearer when setting up config.

Package.json will need reversing to use the original links and name but leave in the updated socket io client version.

### CODE REVIEW

**Comment by isaacvitor** [comment]
Hi @nateainsworth 
I noticed that you did a good job on this PR, thank you! 
But I did not understand why you changed the repository URL to point it to[ your repository](https://github.com/isaacvitor/node-red-contrib-socketio-client/blob/dad19399aea6f95d00aada598bfb19b49c961aab/package.json#L1).

**Comment by nateainsworth** [comment]
All sorted

File: README.md [hunk 1]
```diff
@@ -1,17 +1,29 @@
...
+Original Listener:
 
 > Socket.IO Connector -> Socket.IO Listener -> Payload
-
 ![How to use](https://raw.githubusercontent.com/isaacvitor/generalcontent/master/node-red-contrib-socketio-client/nodered_socketio_ex01.png "How to use")
 
+Listener with callback:
+
+> Socket.IO Connector -> Socket.IO Listener In -> Payload -> Socket.IO Listener Out
+![How to use](https://raw.githubusercontent.com/nateainsworth/Git-docs-images/master/node-red-contrib-socketio-client/listener%20with%20callback.png "How to use")
```
Comment (isaacvitor) [inline README.md:26]: Could you change this?

**Review by isaacvitor — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (nateainsworth) [reply README.md:26]: Sorry, which bit do you want me to remove the images?

Comment (isaacvitor) [reply README.md:26]: this image is important... just to change the URL upload to my repo, please

Comment (nateainsworth) [reply README.md]: All done I've pointed it to your repo, but you'll want to upload the image to your repo download it from here https://github.com/nateainsworth/Git-docs-images/blob/master/node-red-contrib-socketio-client/listener%20with%20callback.png

As I couldn't see the image in your repo https://github.com/isaacvitor/generalcontent/tree/master/node-red-contrib-socketio-client

**Review by isaacvitor — APPROVED** [review]
LGTM

File: README.md (hunk 1 above)
Comment (isaacvitor) [reply README.md]: I didn't upload 👎 sorry

**Pull request merged**